src/EFCore.PG/Metadata/Conventions/NpgsqlValueGenerationConvention.cs

```diff
@@ -59,22 +59,33 @@ public class NpgsqlValueGenerationConvention : RelationalValueGenerationConventi
         }
 
         /// <summary>
-        ///     Returns the store value generation strategy to set for the given property.
+        /// Returns the store value generation strategy to set for the given property.
         /// </summary>
-        /// <param name="property"> The property. </param>
-        /// <returns> The store value generation strategy to set for the given property. </returns>
+        /// <param name="property">The property.</param>
+        /// <returns>The store value generation strategy to set for the given property.</returns>
         protected override ValueGenerated? GetValueGenerated(IConventionProperty property)
-            => GetValueGenerated((IProperty)property);
+        {
+            var tableName = property.DeclaringEntityType.GetTableName();
+            if (tableName == null)
+            {
+                return null;
+            }
+
+            return GetValueGenerated(property, tableName, property.DeclaringEntityType.GetSchema());
+        }
 
         /// <summary>
-        ///     Returns the store value generation strategy to set for the given property.
+        /// Returns the store value generation strategy to set for the given property.
         /// </summary>
-        /// <param name="property"> The property. </param>
-        /// <returns> The store value generation strategy to set for the given property. </returns>
-        public static new ValueGenerated? GetValueGenerated([NotNull] IProperty property)
-            => RelationalValueGenerationConvention.GetValueGenerated(property)
-                ?? (property.GetValueGenerationStrategy() != NpgsqlValueGenerationStrategy.None
-                    ? ValueGenerated.OnAdd
-                    : (ValueGenerated?)null);
+        /// <param name="property">The property.</param>
+        /// <param name="tableName">The table name.</param>
+        /// <param name="schema">The schema.</param>
+        /// <returns>The store value generation strategy to set for the given property.</returns>
+        public static new ValueGenerated? GetValueGenerated(
+            [NotNull] IProperty property, [NotNull] string tableName, [CanBeNull] string schema)
```

Comment:
Weird wrap.

Reply:
Copy-paste from upstream, and also isn't too bad...